crates/cairo-lint-core/src/diagnostics.rs

```diff
@@ -1,33 +1,13 @@
-use annotate_snippets::{Level, Renderer, Snippet};
 use cairo_lang_compiler::db::RootDatabase;
-use cairo_lang_diagnostics::{DiagnosticEntry, Severity};
-use cairo_lang_filesystem::db::FilesGroup;
+use cairo_lang_diagnostics::format_diagnostics as cairo_format_diagnostics;
+use cairo_lang_diagnostics::DiagnosticEntry;
 use cairo_lang_semantic::SemanticDiagnostic;
 use cairo_lang_utils::Upcast;
 
-pub fn format_diagnostic<'a>(
-    diagnostic: &'a SemanticDiagnostic,
-    db: &'a RootDatabase,
-    renderer: &Renderer,
-) -> String {
-    let location = diagnostic.location(db.upcast());
-    let file_id = location.file_id;
-    let span = location.span;
-    let level = match diagnostic.severity() {
-        Severity::Warning => Level::Warning,
-        Severity::Error => Level::Error,
-    };
-    let res = renderer
-        .render(
-            level.title(&diagnostic.format(db)).snippet(
-                Snippet::source(db.file_content(file_id).unwrap().as_ref())
-                    // We give the wole file as string input so the start line is 1
-                    .line_start(1)
-                    .origin(&file_id.full_path(db.upcast()))
-                    .fold(true)
-                    .annotation(level.span(span.to_str_range())),
-            ),
-        )
-        .to_string();
-    format!("{}\n", res)
+pub fn format_diagnostic<'a>(diagnostic: &'a SemanticDiagnostic, db: &'a RootDatabase) -> String {
```

Comment:
It's used only once, i don't see a reason to keep it

Reply:
It's part of the public api of `cairo-lint-core`, and it's used inside scarb's linter cli